server/src/main/java/org/elasticsearch/common/blobstore/BlobPath.java

```diff
@@ -35,12 +35,12 @@ private BlobPath(List<String> paths) {
     }
 
     public BlobPath add(String path) {
-        return new BlobPath(CollectionUtils.appendToCopy(this.paths, path));
+        return new BlobPath(CollectionUtils.appendToCopy(this.paths, path.endsWith("/") ? path.substring(0, path.length() - 1) : path));
```

Comment:
I think it's better to use the `SEPARATOR` constant here instead of `"/"` in place for consistency sake